pkg/operator/ceph/cluster/cluster.go

```diff
@@ -176,66 +179,89 @@ func (c *cluster) doOrchestration(rookImage string, cephVersion cephver.CephVers
 	return nil
 }
 
-func (c *ClusterController) initializeCluster(cluster *cluster, clusterObj *cephv1.CephCluster) {
+func (c *ClusterController) initializeCluster(cluster *cluster, clusterObj *cephv1.CephCluster) error {
 	cluster.Spec = &clusterObj.Spec
 
 	// Check if the dataDirHostPath is located in the disallowed paths list
 	cleanDataDirHostPath := path.Clean(cluster.Spec.DataDirHostPath)
 	for _, b := range disallowedHostDirectories {
 		if cleanDataDirHostPath == b {
 			logger.Errorf("dataDirHostPath (given: %q) must not be used, conflicts with %q internal path", cluster.Spec.DataDirHostPath, b)
-			return
+			return nil
 		}
 	}
 
+	// The Ceph user the operator will use for management
+	cephUser := client.AdminUsername
+
 	// Depending on the cluster type choose the correct orchestation
 	if cluster.Spec.External.Enable {
 		err := c.configureExternalCephCluster(cluster)
 		if err != nil {
 			config.ConditionExport(c.context, c.namespacedName, cephv1.ConditionFailure, v1.ConditionTrue, "ClusterFailure", "Failed to configure external ceph cluster")
-			logger.Errorf("failed to configure external ceph cluster. %v", err)
-			return
+			return errors.Wrap(err, "failed to configure external ceph cluster")
 		}
+		cephUser = cluster.Info.ExternalCred.Username
```

Comment:
You need to use `isExternalHealthCheckUserAdmin(cluster.Info.AdminSecret)`  to know whether we need to use `client.AdminUsername` or `cluster.Info.ExternalCred.Username`.

Reply:
Previously, this code was determining what user should be set:
```
	clientToUse := client.AdminUsername
	if cluster.Spec.External.Enable {
		clientToUse = cluster.Info.ExternalCred.Username
	}
```
My changes basically have this same behavior. If it's an external cluster, always use `cluster.Info.ExternalCred.Username`. 

I see that `populateExternalClusterInfo()` doesn't always populate the external creds if it's the admin user. Shouldn't the external creds be set to the admin user in that case? I'll push a change in a new commit for what I would expect.